AppCenterDataStorage/AppCenterDataStorage/MSDataStore.h

```diff
@@ -16,6 +16,11 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/**
+ * Error domain for Storage.
+ */
+static NSString *const MSDataStorageErrorDomain = @"MSDataStorageErrorDomain";
```

Comment:
Why do we use a new error domain? There is already one defined for AppCenter. I think that we should use it for all modules.

Reply:
Do you want to have a single domain that covers all different errors? What's the benefit of combining all A/C errors to one domain?

Reply:
It's just more appropriate. Cleaner and easier for customers to use. Best to just see if it's an "app center" error instead of having to check for every single module's error domain type. Plus, every time we create a new error domain, we use the wrong format, but the one in the core module is formatted correctly.